python/ray/experimental/channel.py

```diff
@@ -58,6 +58,8 @@ def __init__(
         self,
         buffer_size_bytes: Optional[int] = None,
         num_readers: int = 1,
+        _reader_node_id: Optional[str] = None,
+        _writer_channel: Optional["Channel"] = None,
```

Comment:
I think for usability, it'd be great if we could make as many args non-optional as possible, and then as part of the implementation, we set up the reader/writer channels as needed if other end is remote. That way we don't need to remember to pass the right args / create new channels on the other end. I'm imagining something like this:

```python
# Caller always passes reader_node_id.
# All setup for the remote node handled here.
ch = Channel(num_readers=n, reader_node_id=XXX)

# Pass ch to a remote actor.
reader.pass.remote(ch)

class Reader
  def pass(self, ch):
    # Use ch directly instead of calling Channel(_writer_channel=ch).
    pass
```

Reply:
I am still wrapping this up.